aten/src/ATen/native/native_functions.yaml

```diff
@@ -98,8 +98,10 @@
   dispatch:
     CompositeExplicitAutograd: _fw_primal
 
-- func: _make_dual(Tensor(a) primal, Tensor tangent, int level) -> Tensor(a)
+- func: _make_dual(Tensor(a) self, Tensor tangent, int level) -> Tensor(a)
```

Comment:
Currently functionalization pass codegen seems to assume that first input for view functions must be `self`. Is this intended @bdhirsh?

Reply:
We do have views that don't have self as arg: https://github.com/pytorch/pytorch/blob/6b1d8e5bb20fc56d937f3177f0c3cc6d0d37e0bd/tools/autograd/gen_inplace_or_view_type.py#L71 ?

Reply:
Probably not intended? Could you file an issue please and cc Brian on it?

Reply:
Yep, the functionalization codegen makes some assumptions about the schema of views. It doesn't have to handle views that have a `CompositeImplicitAutograd kernel` though (it just decomposes them instead). Since `_make_dual` is CompositeImplicitAutograd, I wouldn't expect a problem. Is codegen complaining?

Reply:
oh yep, I see this PR makes that op no longer CompositeImplicitAutograd. I'll work on a fix.

Reply:
Actually, I think the codegen can handle this case. Do you see anything failing?

Reply:
If I name the first input of `_make_dual` `primal` instead `self` I run into the following:
```
aten/src/ATen/RegisterFunctionalization_3.cpp: In function ‘at::Tensor at::functionalization::_make_dual(c10::DispatchKeySet, const at::Tensor&, const at::Tensor&, int64_t)’:
aten/src/ATen/RegisterFunctionalization_3.cpp:82:99: error: ‘self’ was not declared in this scope; did you mean ‘c10::attr::self’?
   82 |       auto out = at::functionalization::impl::create_functional_tensor_with_view_meta(tmp_output, self, view_meta);
      |                                                                                                   ^~~~
      |                                                                                                   c10::attr::self

```

In `RegisterFunctionalization_3.cpp` it looks like it is generating a function with signature `at::Tensor _make_dual(c10::DispatchKeySet dispatchKeySet, const at::Tensor & primal, const at::Tensor & tangent, int64_t level)`, but later in the function we call into `at::functionalization::impl::create_functional_tensor_with_view_meta(tmp_output, self, view_meta);` which references `self`. Perhaps we should just the name of the input instead of `self` here?

Reply:
Ah yep, thanks! I'll make a patch.